solidity/contracts/token/extensions/HypERC20RebasingCollateral.sol

```diff
@@ -0,0 +1,105 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity >=0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/extensions/ERC4626.sol";
+import {TokenMessage} from "../libs/TokenMessage.sol";
+import {HypERC20Collateral} from "../HypERC20Collateral.sol";
+import {TypeCasts} from "../../libs/TypeCasts.sol";
+
+/**
+ * @title Hyperlane ERC20 Token Collateral with deposits collateral to a vault
+ * @author ltyu
+ */
+contract HypERC20RebasingCollateral is HypERC20Collateral {
```

Comment:
how does this differ from the existing vault collateral implementation? we should rename for clarity or replace the existing one

Reply:
Even though, the assetsToShares and assetsToShares functions are the same, the synthetic token itself doesn't have an underlying token so it doesn't confer to the 4626 interface. It's a lot of surface area for not a lot of benefit.